### PR TITLE
Bump version to 0.8.0

### PR DESCRIPTION
**Pull request by sunfishcode**
This corresponds to the 0.8.0 packages just published on crates.io. Notably, this includes the `lib` name change from `wasmtime_api` to `wasmtime`. Hopefully the publishing process setttles down now and won't need quite so many changes in the future.

### CODE REVIEW

File: scripts/test-all.sh [hunk 1]
```diff
@@ -95,8 +95,9 @@ if rustup toolchain list | grep -q nightly; then
     cargo +nightly test --features lightbeam
 
     # Also run wasmtime-py and wasmtime-rust's tests.
+    # Temporarily disable wasmtime-py due to https://github.com/bytecodealliance/wasmtime/issues/468
+    #  --package wasmtime-py
     RUST_BACKTRACE=1 cargo +nightly test \
-      --package wasmtime-py \
```
Comment (kubkon) [inline scripts/test-all.sh]: How come this is needed here?

Comment (sunfishcode) [reply scripts/test-all.sh]: See the comment just above; there's currently a bug affecting testing of wasmtime-py.

**Comment by joshtriplett** [comment]
So, shouldn't this be merged verbatim on top of whatever commit it was originally based on, and tagged accordingly, so that we have an exact copy in git of what was uploaded? We can then merge that in, but I don't think we want to just merge this on *top* of the current trunk, as 0.8.0 was already released and that wouldn't match what was released.

**Comment by sunfishcode** [comment]
Yes, I should have merged this earlier. If this is creates trouble for anyone, I'm happy to try to help, but for now I think the answer is just to fix the process for next time.

**Pull request closed without merging**